### ISSUE TITLE
segmentation fault when importing from tomboy folder

### ISSUE THREAD

**Issue by yennor** [giuspen/cherrytree#2152]
**Version, Operative system**
Cherrytree 0.99.51
Debian Testing

**Describe the bug**
When trying to import from tomboy folder I get a 
`66321 segmentation fault (core dumped)  cherrytree`

I've did a strace, deleting the mentioned file doesn-t change anything, it will just segfault on the next one...:

```
ecvmsg(3, {msg_namelen=0}, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(3, {msg_namelen=0}, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=3, events=POLLIN|POLLOUT}], 1, -1) = 1 ([{fd=3, revents=POLLOUT}])
writev(3, [{iov_base="\16*\2\0\320\7@\6", iov_len=8}, {iov_base=NULL, iov_len=0}, {iov_base="", iov_len=0}], 3) = 8
poll([{fd=3, events=POLLIN}], 1, -1)    = 1 ([{fd=3, revents=POLLIN}])
recvmsg(3, {msg_name=NULL, msg_namelen=0, msg_iov=[{iov_base="\1 \3373\0\0\0\0\267\7\0\0\0\0\0\0d\0046\3\0\0\0\0\0\0\0\0\0\0\0\0", iov_len=4096}], msg_iovlen=1, msg_controllen=0, msg_flags=0}, 0) = 32
recvmsg(3, {msg_namelen=0}, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(3, {msg_namelen=0}, 0)          = -1 EAGAIN (Resource temporarily unavailable)
poll([{fd=3, events=POLLIN|POLLOUT}], 1, -1) = 1 ([{fd=3, revents=POLLOUT}])
writev(3, [{iov_base="(*\4\0\320\7@\6\267\7\0\0\0\0\0\0", iov_len=16}, {iov_base=NULL, iov_len=0}, {iov_base="", iov_len=0}], 3) = 16
poll([{fd=3, events=POLLIN}], 1, -1)    = 1 ([{fd=3, revents=POLLIN}])
recvmsg(3, {msg_name=NULL, msg_namelen=0, msg_iov=[{iov_base="\1\1\3403\0\0\0\0\260\"\7\1\316\2N\1\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0\0", iov_len=4096}], msg_iovlen=1, msg_controllen=0, msg_flags=0}, 0) = 32
recvmsg(3, {msg_namelen=0}, 0)          = -1 EAGAIN (Resource temporarily unavailable)
recvmsg(3, {msg_namelen=0}, 0)          = -1 EAGAIN (Resource temporarily unavailable)
write(4, "\1\0\0\0\0\0\0\0", 8)         = 8
write(15, "\1\0\0\0\0\0\0\0", 8)        = 8
futex(0x562a37cb73e0, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7febc78102f0, FUTEX_WAKE_PRIVATE, 1) = 0
write(4, "\1\0\0\0\0\0\0\0", 8)         = 8
futex(0x562a37483810, FUTEX_WAKE_PRIVATE, 2147483647) = 0
futex(0x562a373c3190, FUTEX_WAKE_PRIVATE, 1) = 1
newfstatat(AT_FDCWD, "/", {st_mode=S_IFDIR|0755, st_size=4096, ...}, AT_SYMLINK_NOFOLLOW) = 0
openat(AT_FDCWD, "/proc/self/mountinfo", O_RDONLY|O_CLOEXEC) = 16
newfstatat(16, "", {st_mode=S_IFREG|0444, st_size=0, ...}, AT_EMPTY_PATH) = 0
read(16, "22 29 0:20 / /sys rw,nosuid,node"..., 1024) = 1024
readlink("/proc", 0x7ffde2bf5760, 1023) = -1 EINVAL (Invalid argument)
readlink("/proc/self", "68402", 1023)   = 5
readlink("/proc/68402", 0x7ffde2bf5760, 1023) = -1 EINVAL (Invalid argument)
readlink("/proc/68402/mountinfo", 0x7ffde2bf5760, 1023) = -1 EINVAL (Invalid argument)
read(16, "c,relatime shared:10 - pstore ps"..., 1024) = 1024
read(16, "time shared:21 - ramfs ramfs rw,"..., 1024) = 1024
read(16, "id=1001\n694 565 0:52 / /run/user"..., 1024) = 1024
read(16, "aid rw,lowerdir=/home/rmm/.confi"..., 1024) = 519
read(16, "", 1024)                      = 0
close(16)                               = 0
getuid()                                = 1001
newfstatat(AT_FDCWD, "/run/mount/utab", {st_mode=S_IFREG|0644, st_size=111, ...}, 0) = 0
openat(AT_FDCWD, "/run/mount/utab", O_RDONLY|O_CLOEXEC) = 16
newfstatat(16, "", {st_mode=S_IFREG|0644, st_size=111, ...}, AT_EMPTY_PATH) = 0
read(16, "SRC=/dev/mapper/luks-cacda2c2-9b"..., 4096) = 111
read(16, "", 4096)                      = 0
close(16)                               = 0
write(9, "\1\0\0\0\0\0\0\0", 8)         = 8
futex(0x562a374cfd40, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7febb8017f60, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x562a373e25b8, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x562a374c9120, FUTEX_WAKE_PRIVATE, 1) = 0
write(9, "\1\0\0\0\0\0\0\0", 8)         = 8
futex(0x562a374cfd40, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7febb8017f60, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x562a373e25b8, FUTEX_WAKE_PRIVATE, 1) = 1
write(9, "\1\0\0\0\0\0\0\0", 8)         = 8
futex(0x562a373e25b8, FUTEX_WAKE_PRIVATE, 1) = 1
write(9, "\1\0\0\0\0\0\0\0", 8)         = 8
futex(0x562a374cfd40, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7febc78102f0, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x7febc78102f0, FUTEX_WAKE_PRIVATE, 1) = 0
inotify_rm_watch(11, 4)                 = 0
futex(0x7febc7fff408, FUTEX_WAKE_PRIVATE, 1) = 1
openat(AT_FDCWD, "/home/rmm/.local/share/tomboy", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 14
newfstatat(14, "", {st_mode=S_IFDIR|0700, st_size=20480, ...}, AT_EMPTY_PATH) = 0
lseek(14, 0, SEEK_SET)                  = 0
getdents64(14, 0x562a37d30930 /* 228 entries */, 32768) = 14488
getdents64(14, 0x562a37d30930 /* 0 entries */, 32768) = 0
close(14)                               = 0
newfstatat(AT_FDCWD, "/home/rmm/.local/share/tomboy/af031727-aad3-4f34-a0a4-4f189b9600b6.note", {st_mode=S_IFREG|0700, st_size=2473, ...}, 0) = 0
newfstatat(AT_FDCWD, "/home/rmm/.local/share/tomboy/af031727-aad3-4f34-a0a4-4f189b9600b6.note", {st_mode=S_IFREG|0700, st_size=2473, ...}, 0) = 0
newfstatat(AT_FDCWD, "/home/rmm/.local/share/tomboy/af031727-aad3-4f34-a0a4-4f189b9600b6.note", {st_mode=S_IFREG|0700, st_size=2473, ...}, 0) = 0
newfstatat(AT_FDCWD, "/home/rmm/.local/share/tomboy/af031727-aad3-4f34-a0a4-4f189b9600b6.note", {st_mode=S_IFREG|0700, st_size=2473, ...}, 0) = 0
openat(AT_FDCWD, "/home/rmm/.local/share/tomboy/af031727-aad3-4f34-a0a4-4f189b9600b6.note", O_RDONLY) = 14
lseek(14, 0, SEEK_CUR)                  = 0
read(14, "\357\273\277<?xml version=\"1.0\" encoding="..., 8192) = 2473
read(14, "", 5719)                      = 0
close(14)                               = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x8} ---
+++ killed by SIGSEGV (core dumped) +++

```

**To Reproduce**
1. Go to import/tomboy folder
2. import
3. seg fault

I-ve added a file, rename it to .note
[6b638d9c-494d-44dd-8d3f-6592ca8dc080.txt](https://github.com/giuspen/cherrytree/files/9983263/6b638d9c-494d-44dd-8d3f-6592ca8dc080.txt)

Also when specifiing the .config/tomboy folder instead of .local/share/tomboy, it will produce the same segfault when trying to read manifest.xml

**Comment by giuspen** [comment]
I can reproduce thanks

**Comment by giuspen** [comment]
will be fixed in 0.99.52
